icubam/backoffice/handlers/icus.py

```diff
@@ -31,29 +31,37 @@ def get(self):
 class ICUHandler(base.BaseHandler):
   ROUTE = "/icu"
 
-  @tornado.web.authenticated
-  def get(self):
-    icu = self.db.get_icu(self.get_query_argument('id', None))
-    icu = icu if icu is not None else store.ICU()
-    if icu.is_active is None:
-      icu.is_active = True
-
+  def do_render(self, icu, error=False):
     if self.user.is_admin:
       regions = self.db.get_regions()
     if not self.user.is_admin:
       regions = [e.region for e in self.user.managed_icus]
     regions.sort(key=lambda r: r.name)
-    self.render("icu.html", icu=icu, regions=regions, error='')
+
+    icu = icu if icu is not None else store.ICU()
+    if icu.is_active is None:
+      icu.is_active = True
+    return self.render("icu.html", icu=icu, regions=regions, error=error)
 
   @tornado.web.authenticated
-  def post(self):
+  def get(self):
     icu = self.db.get_icu(self.get_query_argument('id', None))
+    return self.do_render(icu)
+
+  @tornado.web.authenticated
+  def post(self):
     values = self.parse_from_body(store.ICU)
-    values["is_active"] = bool(values["is_active"])
-    # TODO(olivier): try catch error here.
-    if icu is None:
-      self.db.add_icu(self.user.user_id, store.ICU(**values))
-    else:
-      self.db.update_icu(self.user.user_id, icu.icu_id, values)
+    values["is_active"] = values["is_active"] == 'True'
```

Comment:
make sure this is not "on"
(apparently it depends on the checkbox component you use, but the one used in the users page gives:
empty string when false
"on" when true

I usually do print(self.request.body) to debug how the values arrive in the handler.

Reply:
good point. but because of line 44 and the component has a value in icu.html, it comes True / False as string